logictest: memory budget exceeded in TestLogic_udf

pkg/sql/logictest/tests/local-legacy-schema-changer/local-legacy-schema-changer_test.TestLogic_udf [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/10060771?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/10060771?buildTab=artifacts#/) on master @ [0036f11ca58e2ab1eeef0d785dbe4d8641e8cba3](https://github.com/cockroachdb/cockroach/commits/0036f11ca58e2ab1eeef0d785dbe4d8641e8cba3):


```
=== RUN   TestLogic_udf
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/7f1e7ac178916dd9f085ac9bb5e76ebe/logTestLogic_udf1196170223
    test_log_scope.go:79: use -show-logs to present logs inline
[00:18:05] setting distsql_workmem='43291B';
=== CONT  TestLogic_udf
    logic.go:3995: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/12168/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-legacy-schema-changer/local-legacy-schema-changer_test_/local-legacy-schema-changer_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/udf:3465: error while processing
    logic.go:3995: test was successful but validation upon completion failed: dropping database test failed: pq: polling for queued jobs to complete: poll-show-jobs: root: memory budget exceeded: 10240 bytes requested, 201318400 currently allocated, 201326592 bytes in budget
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/7f1e7ac178916dd9f085ac9bb5e76ebe/logTestLogic_udf1196170223
--- FAIL: TestLogic_udf (194.74s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_udf.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-27896

--- a/pkg/sql/logictest/logic.go
+++ b/pkg/sql/logictest/logic.go
@@ -1326,7 +1326,7 @@ func (t *logicTest) newCluster(
 	if serverArgs.MaxSQLMemoryLimit == 0 {
 		// Specify a fixed memory limit (some test cases verify OOM conditions;
 		// we don't want those to take long on large machines).
-		serverArgs.MaxSQLMemoryLimit = 192 * 1024 * 1024
+		serverArgs.MaxSQLMemoryLimit = 256 * 1024 * 1024
 	}
 	// We have some queries that bump into 100MB default temp storage limit
 	// when run with fakedist-disk config, so we'll use a larger limit here.

